nntrainer/layers/layer_node.h

```diff
@@ -157,6 +158,105 @@ class LayerNode : public ml::train::Layer, public GraphNode {
    */
   ActivationType getActivationType();
 
+  /**
+   * @brief     Get number of inputs
+   * @retval    number of inputs
+   */
+  unsigned int getNumInputs() const { return input_layers.size(); }
+
+  /**
+   * @brief     Get number of outputs
+   * @retval    number of outputs
+   */
+  unsigned int getNumOutputs() const { return output_layers.size(); }
+
+  /**
+   * @brief Get the Input Layers object
+   *
+   * @return const std::vector<std::string>&
+   */
+  const std::vector<std::string> &getInputLayers() const {
+    return input_layers;
+  }
+
+  /**
+   * @brief Get the Output Layers object
+   *
+   * @return const std::vector<std::string>&
+   */
+  const std::vector<std::string> &getOutputLayers() const {
+    return output_layers;
+  }
+
+  /**
+   * @brief Update input layers entry name
+   *
+   * @param from The name to be updated
+   * @param to The name to be updated to
+   */
+  void updateInputLayers(const std::string &from, const std::string &to);
+
+  /**
+   * @brief Update the input layers name at the given idx
+   *
+   * @param idx The index at which layer name must be updated
+   * @param to The name to be updated to
+   */
+  void updateInputLayers(const unsigned int idx, const std::string &to);
+
+  /**
+   * @brief Add name to the input layers
+   *
+   * @param in_layer Name to be added
+   */
+  void addInputLayers(const std::string &in_layer) {
+    input_layers.push_back(in_layer);
+    layer->setNumInputs(input_layers.size());
+  }
+
+  /**
+   * @brief Add name to the output layers
+   *
+   * @param out_layer Name to be added
+   */
+  void addOutputLayers(const std::string &out_layer) {
+    output_layers.push_back(out_layer);
+    layer->setNumOutputs(output_layers.size());
+  }
+
+  /**
+   * @brief Set the Input Layers object
+   *
+   * @param layers Name of the layers
+   */
+  void setInputLayers(const std::vector<std::string> &layers) {
+    input_layers = layers;
+    layer->setNumInputs(layers.size());
+  }
+
+  /**
+   * @brief Set the Output Layers object
+   *
+   * @param layers Name of the layers
+   */
+  void setOutputLayers(const std::vector<std::string> &layers) {
+    output_layers = layers;
+    layer->setNumOutputs(layers.size());
+  }
+
+  /**
+   * @brief this function helps exporting the layer in a predefined format,
+   * while workarounding issue caused by templated function type eraser
+   *
+   * @param exporter exporter that conatins exporting logic
+   * @param method enum value to identify how it should be exported to
+   */
+  virtual void
+  export_to(Exporter &exporter,
+            ExportMethods method = ExportMethods::METHOD_STRINGVECTOR) const {
+    exporter.saveResult(props, method, this);
```

Comment:
Maybe call Layer::export_to in this stage?

Reply:
Updated with a new commit, as interpreter codes are also changed to use LayerNode than Layer.  Please check once.

Reply:
Checked. Looks good to me :)